### PR TITLE
Added stackpath dns

### PR DESCRIPTION
**Pull request by brentahughes**
This is pending the approval and merge of https://github.com/xenolf/lego/pull/660

### CODE REVIEW

**Comment by brentahughes** [comment]
The dependent PR has now been merged to master so this PR can continue forward

**Comment by mholt** [comment]
Thanks for your contribution!

Coincidentally, changes are happening in the core of Caddy right now that could potentially deprecate DNS provider plugins; they would no longer be needed, and instead, [all of lego's DNS providers](https://github.com/xenolf/lego/tree/master/providers/dns) would be supported by default for every Caddy instance without the need for plugins. (The downside is the Caddy binary will grow in size.) 

There's a poll taking place here: https://caddy.community/t/poll-we-can-end-dns-provider-plugins-should-we/4887?u=matt - I would like to get your opinion!

**Comment by mholt** [comment]
Would you please be able to update this to be compatible with the latest changes in lego (v2.0.1)? Be sure to test it, then I can merge this in. :) Thanks!

**Comment by brentahughes** [comment]
Sorry for the delay. This has been updated

File: stackpath/stackpath.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+func NewDNSProvider(credentials ...string) (caddytls.ChallengeProvider, error) {
+	switch len(credentials) {
+	case 0:
+		return stackpath.NewDNSProvider()
+	case 3:
+		config := stackpath.NewDefaultConfig()
+		config.ClientID = credentials[0]
+		config.ClientSecret = credentials[1]
+		config.StackID = credentials[2]
+		return stackpath.NewDNSProviderConfig()
```
Comment (mholt) [inline stackpath/stackpath.go]: Is `config` even used?

Comment (brentahughes) [reply stackpath/stackpath.go]: Nope. I'm dumb. I will get this updated with proper testing results later today.

**Comment by brentahughes** [comment]
Added unit tests with the output

```
$ go test -v -cover ./stackpath/...
=== RUN   TestNewDNSProvider
=== RUN   TestNewDNSProvider/no_creds_given
=== RUN   TestNewDNSProvider/creds_given
=== RUN   TestNewDNSProvider/bad_cred_count
--- PASS: TestNewDNSProvider (0.00s)
    --- PASS: TestNewDNSProvider/no_creds_given (0.00s)
    --- PASS: TestNewDNSProvider/creds_given (0.00s)
    --- PASS: TestNewDNSProvider/bad_cred_count (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/caddyserver/dnsproviders/stackpath	(cached)	coverage: 100.0% of statements
```

**Review by mholt — APPROVED** [review]
Much better, thanks! And nice job with the tests.

**Pull request merged**